### PR TITLE
[Quantum RPG] Finish Hadamard Hills Zone

### PR DESCRIPTION
**Pull request by dstrain115**
- Add room descriptions and encounters.

### CODE REVIEW

File: unitary/examples/quantum_rpg/final_state_preparation/hadamard_hills.py [hunk 1]
```diff
@@ -18,23 +18,82 @@
...
+
+CHARLES = Item(
+    keyword_actions=[
+        (
+            TALK,
+            ["man", "charles", "elder", "physicist"],
+            (
+                "The physicist introduces himself as Charles.  'I am working on a\n"
+                "device to communicate in the quantum realm.  I am currently testing\n"
+                "a sensitive receiver detecting a laser beam from across the hills.\n"
```
Comment (cognigami) [inline unitary/examples/quantum_rpg/final_state_preparation/hadamard_hills.py:35]: REPLACE detecting
WITH to detect

File: unitary/examples/quantum_rpg/final_state_preparation/hadamard_hills.py [hunk 2]
```diff
@@ -18,23 +18,82 @@
...
+                "The communication is immune to eavesdropping, but the receiver has\n"
+                "to measure in either the computational basis (I) or the Hadamard (H)\n"
+                "basis, and it cannot know which basis the laser beam was sent in.\n"
```
Comment (cognigami) [inline unitary/examples/quantum_rpg/final_state_preparation/hadamard_hills.py:38]: REPLACE basis, and it
WITH basis. The receiver

File: unitary/examples/quantum_rpg/final_state_preparation/hadamard_hills.py [hunk 3]
```diff
@@ -18,23 +18,82 @@
...
+    keyword_actions=[
+        (
+            TALK,
+            ["man", "charles", "elder", "physicist"],
+            (
+                "The physicist introduces himself as Charles.  'I am working on a\n"
+                "device to communicate in the quantum realm.  I am currently testing\n"
+                "a sensitive receiver detecting a laser beam from across the hills.\n"
+                "The communication is immune to eavesdropping, but the receiver has\n"
+                "to measure in either the computational basis (I) or the Hadamard (H)\n"
```
Comment (cognigami) [inline unitary/examples/quantum_rpg/final_state_preparation/hadamard_hills.py:37]: Does the player always encounter Charles before Gilles? If not, they might not know that by "computational basis" you mean Identity.

File: unitary/examples/quantum_rpg/final_state_preparation/hadamard_hills.py [hunk 4]
```diff
@@ -115,21 +174,32 @@
...
     ),
     Location(
         label="hadamard6",
         title="Within the Foothills",
-        description=("Room description forthcoming."),
-        encounters=[],
+        description=(
+            "You are entering a cluster of hills that extend to the north.\n"
+            "To the east and west, two identical peaks rise from the maze\n"
+            "of foothills.  The mirroring of terrain on both sides of you\n"
```
Comment (cognigami) [inline unitary/examples/quantum_rpg/final_state_preparation/hadamard_hills.py:186]: DEL of you

File: unitary/examples/quantum_rpg/final_state_preparation/hadamard_hills.py [hunk 5]
```diff
@@ -190,89 +284,150 @@
     Location(
         label="hadamard11_0",
         title="East Hut",
-        description=("Room description forthcoming."),
+        description=(
+            "You have entered an austere room filled with neatly organized\n"
+            "booksheles and filing cabinets.  In one corner, a carefully set\n"
```
Comment (cognigami) [inline unitary/examples/quantum_rpg/final_state_preparation/hadamard_hills.py:289]: DEL a carefully set

File: unitary/examples/quantum_rpg/final_state_preparation/hadamard_hills.py [hunk 6]
```diff
@@ -190,89 +284,150 @@
...
+            "up optical table is set up with some kind of photonics experiment.\n"
```
Comment (cognigami) [inline unitary/examples/quantum_rpg/final_state_preparation/hadamard_hills.py:290]: REPLACE up
WITH an
INSERT carefully
BETWEEN up, with

File: unitary/examples/quantum_rpg/final_state_preparation/hadamard_hills.py [hunk 7]
```diff
@@ -190,89 +284,150 @@
...
+        items=[CHARLES, EAST_EGG],
         exits={Direction.NORTH: "hadamard10_0"},
     ),
     Location(
         label="hadamard11_1",
         title="West Hut",
-        description=("Room description forthcoming."),
+        description=(
+            "You have entered a cozy living space filled to the brim with\n"
+            "papers and books stacked all around the room.  A large pile\n"
```
Comment (cognigami) [inline unitary/examples/quantum_rpg/final_state_preparation/hadamard_hills.py:302]: REPLACE pile
WITH tower

...or some other noun: heap, stack, mound, clump, mountain, lump, assortment

File: unitary/examples/quantum_rpg/final_state_preparation/hadamard_hills.py [hunk 8]
```diff
@@ -190,89 +284,150 @@
...
+            "of hastily piled electronics clutters up a corner of the room.\n"
+            "A desk with a lamp piled high with manuscripts and technical\n"
+            "specifications takes up a big art of the room."
```
Comment (cognigami) [inline unitary/examples/quantum_rpg/final_state_preparation/hadamard_hills.py:305]: REPLACE art
WITH part

File: unitary/examples/quantum_rpg/final_state_preparation/hadamard_hills.py [hunk 9]
```diff
@@ -190,89 +284,150 @@
...
+        items=[GILLES, WEST_EGG],
         exits={Direction.NORTH: "hadamard10_1"},
     ),
     Location(
         label="hadamard12_0",
         title="Upward Slope",
-        description=("Room description forthcoming."),
-        encounters=[],
+        description=(
+            "The trail continues upwards here, towards the top of\n"
```
Comment (cognigami) [inline unitary/examples/quantum_rpg/final_state_preparation/hadamard_hills.py:315]: DEL here

File: unitary/examples/quantum_rpg/final_state_preparation/hadamard_hills.py [hunk 10]
```diff
@@ -190,89 +284,150 @@
...
+        encounters=[purple_foam(3, 0.2)],
         exits={Direction.WEST: "hadamard10_0", Direction.UP: "hadamard13_0"},
     ),
     Location(
         label="hadamard12_1",
         title="Upward Slope",
-        description=("Room description forthcoming."),
-        encounters=[],
+        description=(
+            "The trail continues upwards here, towards the top of\n"
```
Comment (cognigami) [inline unitary/examples/quantum_rpg/final_state_preparation/hadamard_hills.py:327]: DEL here

File: unitary/examples/quantum_rpg/final_state_preparation/hadamard_hills.py [hunk 11]
```diff
@@ -190,89 +284,150 @@
...
         exits={Direction.DOWN: "hadamard13_1", Direction.NORTH: "hadamard15_1"},
     ),
     Location(
         label="hadamard15_0",
         title="East Hill Pavillion",
-        description=("Room description forthcoming."),
+        description=(
+            "At the top of the east hill is a concrete slab.\n"
+            "A post at each of the four corners supports a taut piece\n"
+            "of havey fabric that proides some shade for those that\n"
```
Comment (cognigami) [inline unitary/examples/quantum_rpg/final_state_preparation/hadamard_hills.py:387]: REPLACE havey
WITH heavy

File: unitary/examples/quantum_rpg/final_state_preparation/hadamard_hills.py [hunk 12]
```diff
@@ -190,89 +284,150 @@
...
         exits={Direction.SOUTH: "hadamard14_0"},
     ),
     Location(
         label="hadamard15_1",
         title="West Hill Pavillion",
-        description=("Room description forthcoming."),
+        description=(
+            "At the top of the west hill is a concrete slab.\n"
+            "A post at each of the four corners supports a taut piece\n"
+            "of havey fabric that proides some shade for those that\n"
```
Comment (cognigami) [inline unitary/examples/quantum_rpg/final_state_preparation/hadamard_hills.py:400]: REPLACE havey
WITH heavy

File: unitary/examples/quantum_rpg/final_state_preparation/hadamard_hills.py [hunk 13]
```diff
@@ -190,89 +284,150 @@
...
     Location(
         label="hadamard17",
-        title="Door of the Research Institute",
-        description=("Room description forthcoming."),
+        title="Research Facility Entrance",
+        description=(
+            "As you near the imposing grey buildings, you can see that\n"
+            "they form a research outpost.  What must have once been\n"
+            "an impressive and state-of-the-art facility now shows\n"
+            "signs of damage Pieces of the building have been turned\n"
```
Comment (cognigami) [inline unitary/examples/quantum_rpg/final_state_preparation/hadamard_hills.py:427]: INSERT .
BETWEEN damage, Pieces

File: unitary/examples/quantum_rpg/final_state_preparation/hadamard_hills.py [hunk 14]
```diff
@@ -190,89 +284,150 @@
...
+            "into the iridiscent staic of quantum errors. Broken windows\n"
```
Comment (cognigami) [inline unitary/examples/quantum_rpg/final_state_preparation/hadamard_hills.py:428]: REPLACE staic
WITH static

**Review by cognigami — CHANGES_REQUESTED** [review]

File: unitary/examples/quantum_rpg/final_state_preparation/hadamard_hills.py (hunk 2 above)
Comment (dstrain115) [reply unitary/examples/quantum_rpg/final_state_preparation/hadamard_hills.py]: Done.

File: unitary/examples/quantum_rpg/final_state_preparation/hadamard_hills.py (hunk 4 above)
Comment (dstrain115) [reply unitary/examples/quantum_rpg/final_state_preparation/hadamard_hills.py]: Done.

File: unitary/examples/quantum_rpg/final_state_preparation/hadamard_hills.py (hunk 5 above)
Comment (dstrain115) [reply unitary/examples/quantum_rpg/final_state_preparation/hadamard_hills.py]: Done.

File: unitary/examples/quantum_rpg/final_state_preparation/hadamard_hills.py (hunk 8 above)
Comment (dstrain115) [reply unitary/examples/quantum_rpg/final_state_preparation/hadamard_hills.py]: Done.

File: unitary/examples/quantum_rpg/final_state_preparation/hadamard_hills.py (hunk 10 above)
Comment (dstrain115) [reply unitary/examples/quantum_rpg/final_state_preparation/hadamard_hills.py]: Done.

**Review by dstrain115 — COMMENTED** [review]
As usual, thanks for catching all my typos and grammar issues!

**Pull request merged**